### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by seperalta22**
# OOP school library: add Person, Student and Teacher classes

## Summary

This is the first milestone of the 'OOP school library' project. It has three classes: Person, Student, and Teacher.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    @parent_permission = parent_permission
+  end
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
```
Comment (anagudelogu) [inline person.rb]: - [ ] You did a great job working on this milestone! 💯 However, according to the requirements:

image

The method `of_age` should be private. Please consider applying this! 🤓 🔧

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require relative 'person'
+
+class Student < Person
+  attr_accessor :classroom
+
+  def initialize(classroom:, id: rand(1..1000), name: 'Unknown', age: nil, parent_permission: true)
+    super(id, name, age, parent_permission)
+    @classroom = classroom
+  end
```
Comment (anagudelogu) [inline student.rb]: - [ ] Please be extra careful when you're using named parameters like `parent_permission`. If you look closely, your super function, in this case, `Person` expects the `parent_permission` parameter as a key: value. So the correct way to send the `parent_permission` in your call to super() would be:

```suggestion
  def initialize(classroom:, id: rand(1..1000), name: 'Unknown', age: nil, parent_permission: true)
    super(id, name, age, parent_permission: parent_permission)
    @classroom = classroom
  end
```

Please consider applying this change! 🤓 🔧

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+require_relative 'person'
+
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(specialization, id = rand(1..1000), name = 'Unknown', age = nil, parent_permission: true)
+    super(id, name, age, parent_permission)
+    @specialization = specialization
+  end
```
Comment (anagudelogu) [inline teacher.rb]: - [ ] The same happens here, please consider fixing it! 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @SergioPeralta22, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+require relative 'person'
+
```
Comment (Gambit142) [inline student.rb]: - [OPTIONAL] The correct syntax is `require_relative` and not `require relative`. If you use the latter, your application will crash and you will get this sort of error:

image

Kindly use the correct syntax 💻 🖱️

**Review by Gambit142 — APPROVED** [review]
Hi @SergioPeralta22 👋,

## Status: Approved 🟢   

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

### Highlights
- All linter checks passed ✔️ 
- Created `Person` class ✔️ 
- Created `Teacher` Class ✔️ 
- Created `Student` class ✔️ 

### Optional Suggestion

- [ ] Kindly consider the comment tagged _**[OPTIONAL]**_ below the review summary. 👍 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**